library/Opus/Search/Query.php

```diff
@@ -0,0 +1,479 @@
+	 * @param Opus_Search_Filter_Base $subFilter filter to be satisfied by all matching documents in addition
+	 * @return $this fluent interface
+	 */
+	public function setSubFilter( $name, Opus_Search_Filter_Base $subFilter ) {
+		if ( !is_string( $name ) || !$name ) {
+			throw new InvalidArgumentException( 'invalid sub filter name' );
+		}
+
+		if ( !is_array( $this->_data['subfilters'] ) ) {
+			$this->_data['subfilters'] = array( $name => $subFilter );
+		} else {
+			$this->_data['subfilters'][$name] = $subFilter;
+		}
+
+		return $this;
+	}
+
+	/**
+	 * Removes some previously defined subfilter from current query again.
+	 *
+	 * @note This isn't affecting server-side caching of selected filter but
+	 *       reverting some parts of query compiled on client-side.
+	 *
+	 * @see Opus_Search_Query::setSubFilter()
+	 *
+	 * @param string $name name of filter to remove from query again
+	 * @return $this fluent interface
+	 */
+	public function removeSubFilter( $name ) {
+		if ( !is_string( $name ) || !$name ) {
+			throw new InvalidArgumentException( 'invalid sub filter name' );
+		}
+
+		if ( is_array( $this->_data['subfilters'] ) ) {
+			if ( array_key_exists( $name, $this->_data['subfilters'] ) ) {
+				unset( $this->_data['subfilters'][$name] );
+			}
+
+			if ( !count( $this->_data['subfilters'] ) ) {
+				$this->_data['subfilters'] = null;
+			}
+		}
+
+		return $this;
+	}
+
+	/**
+	 * Retrieves named map of subfilters to include on querying search engine.
+	 *
+	 * @return Opus_Search_Filter_Base[]
+	 */
+	public function getSubFilters() {
+		return $this->_data['subfilters'];
+	}
+
+	public static function getParameterDefault( $name, $fallbackIfMissing, $oldName = null ) {
+		$config   = Opus_Search_Config::getDomainConfiguration();
+		$defaults = $config->parameterDefaults;
+
+		if ( $defaults instanceof Zend_Config ) {
+			return $defaults->get( $name, $fallbackIfMissing );
+		}
+
+		if ( $oldName ) {
+			return $config->get( $oldName, $fallbackIfMissing );
+		}
+
+		return $fallbackIfMissing;
+	}
+
+	/**
+	 * Retrieves configured default offset for paging results.
+	 *
+	 * @return int
+	 */
+	public static function getDefaultStart() {
+		return static::getParameterDefault( 'start', 0 );
+	}
+
+	/**
+	 * Retrieves configured default number of rows to show (per page).
+	 *
+	 * @return int
+	 */
+	public static function getDefaultRows() {
+		return static::getParameterDefault( 'rows', 10, 'numberOfDefaultSearchResults' );
+	}
+
+	/**
+	 * Retrieves configured default sorting.
+	 *
+	 * @return string[]
+	 */
+	public static function getDefaultSorting() {
+		$sorting = static::getParameterDefault( 'sortField', 'score desc' );
+
+		$parts = preg_split( '/[\s,]+/', trim( $sorting ), null, PREG_SPLIT_NO_EMPTY );
+
+		$sorting = array( array_shift( $parts ) );
+
+		if ( !count( $parts ) ) {
+			$sorting[] = 'asc';
+		} else {
+			$dir = array_shift( $parts );
+			if ( strcasecmp( $dir, 'asc' ) || strcasecmp( $dir, 'desc' ) ) {
```

Comment:
This line seems to cause that instead of 'desc' as default direction 'asc' is used. I am not sure what the method is trying to do.